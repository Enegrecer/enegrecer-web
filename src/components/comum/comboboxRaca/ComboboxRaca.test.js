import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ComboboxRaca from '../comboboxRaca';

describe('Combobox Raça', () => {
  it('deve conter todas as raças quando somenteRacasVitima é false ', () => {
    const onChange = jest.fn();
    const comboboxRacaTestemunha = mount(<ComboboxRaca id={'raca-testemunha'} onChange={onChange} classes={'racas'} somenteRacasVitima={false} />);
    const comboboxRacaTestemunhaTextoJson = toJson(comboboxRacaTestemunha);
    expect(comboboxRacaTestemunhaTextoJson).toMatchSnapshot();
  });

  it('deve conter somente as raças Pardo e Negro quando somenteRacasVitima é true ', () => {
    const onChange = jest.fn();
    const comboboxRacaVitima = mount(<ComboboxRaca id={'raca-vitma'} onChange={onChange} classes={'racas'} somenteRacasVitima />);
    const comboboxRacaVitimaTextoJson = toJson(comboboxRacaVitima);
    expect(comboboxRacaVitimaTextoJson).toMatchSnapshot();
  });
});

