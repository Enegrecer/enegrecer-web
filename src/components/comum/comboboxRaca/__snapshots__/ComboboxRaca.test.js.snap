// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Combobox Raça deve conter somente as raças Pardo e Negro quando somenteRacasVitima é true  1`] = `
<comboRaca
  classes="racas"
  id="raca-vitma"
  onChange={[Function]}
  somenteRacasVitima={true}
>
  <combobox
    divClasse="racas"
    id="raca-vitma"
    itens={
      Array [
        "Preta",
        "Parda",
      ]
    }
    label="Selecione a Raça:"
    onChange={[Function]}
    valorPadrao="Selecione a Raça:"
  >
    <div
      className="racas"
    >
      <select
        id="raca-vitma"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Selecione a Raça:
        </option>
        <option
          key="Preta"
          value="Preta"
        >
           
          Preta
           
        </option>
        <option
          key="Parda"
          value="Parda"
        >
           
          Parda
           
        </option>
      </select>
      <label
        htmlFor="raca-vitma"
      >
        Selecione a Raça:
      </label>
    </div>
  </combobox>
</comboRaca>
`;

exports[`Combobox Raça deve conter todas as raças quando somenteRacasVitima é false  1`] = `
<comboRaca
  classes="racas"
  id="raca-testemunha"
  onChange={[Function]}
  somenteRacasVitima={false}
>
  <combobox
    divClasse="racas"
    id="raca-testemunha"
    itens={
      Array [
        "Amarela",
        "Branco",
        "Índigena",
        "Parda",
        "Preta",
      ]
    }
    label="Selecione a Raça:"
    onChange={[Function]}
    valorPadrao="Selecione a Raça:"
  >
    <div
      className="racas"
    >
      <select
        id="raca-testemunha"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Selecione a Raça:
        </option>
        <option
          key="Amarela"
          value="Amarela"
        >
           
          Amarela
           
        </option>
        <option
          key="Branco"
          value="Branco"
        >
           
          Branco
           
        </option>
        <option
          key="Índigena"
          value="Índigena"
        >
           
          Índigena
           
        </option>
        <option
          key="Parda"
          value="Parda"
        >
           
          Parda
           
        </option>
        <option
          key="Preta"
          value="Preta"
        >
           
          Preta
           
        </option>
      </select>
      <label
        htmlFor="raca-testemunha"
      >
        Selecione a Raça:
      </label>
    </div>
  </combobox>
</comboRaca>
`;
